src/auxiliary/mod.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module is about interaction with auxiliary block chain.
+
+use futures::Future;
+use rlp;
+use sha3::{Digest, Keccak256};
+use web3::contract::Options;
+use web3::types::{Address, H160, H256};
+
+use super::ethereum::Ethereum;
+use ethereum::types::{Block, Error, ErrorKind};
+
+const REPORT_BLOCK_ESTIMATED_GAS: i32 = 3_000_000;
```

Comment:
Maybe add a comment where this number comes from?

Reply:
OK, I have added a comment, but in future, there will be more such constant for different contract calls. This will eventually change to externalize config.

Reply:
Follow comment is added 
`This is gas consumed by report block operation. This is calculated by estimate gas
 function from web3.`